x-pack/plugins/actions/server/lib/action_executor.ts

```diff
@@ -97,64 +118,283 @@ export class ActionExecutor {
+    const namespace = spaceId && spaceId !== 'default' ? { namespace: spaceId } : {};
+
+    return await this.executeHelper({
+      actionExecutionId,
+      actionId,
+      ensureAuthorizedFn: async (connectorTypeId: string) => {
+        // We don't allow execute system actions in unsecured manner because they require a request
+        if (actionTypeRegistry.isSystemActionType(connectorTypeId)) {
+          const errorMessage = `Cannot execute unsecured system action`;
+          throw new ActionExecutionError(errorMessage, ActionExecutionErrorReason.Authorization, {
+            actionId,
+            status: 'error',
+            message: errorMessage,
+            retry: false,
+            errorSource: TaskErrorSource.USER,
+          });
+        }
+      },
+      executeLabel: `execute_unsecured_action`,
+      namespace,
+      params,
+      relatedSavedObjects,
+      services,
+      source,
+      spaceId,
+    });
+  }
+
+  public async logCancellation<Source = unknown>({
+    actionId,
     request,
+    relatedSavedObjects,
     source,
-    isEphemeral,
+    executionId,
     taskInfo,
+    consumer,
+    actionExecutionId,
+  }: {
+    actionId: string;
+    actionExecutionId: string;
+    request: KibanaRequest;
+    taskInfo?: TaskInfo;
+    executionId?: string;
+    relatedSavedObjects: RelatedSavedObjects;
+    source?: ActionExecutionSource<Source>;
+    consumer?: string;
+  }) {
+    const { spaces, eventLogger } = this.actionExecutorContext!;
+
+    const spaceId = spaces && spaces.getSpaceId(request);
+    const namespace = spaceId && spaceId !== 'default' ? { namespace: spaceId } : {};
+
+    if (!this.actionInfo || this.actionInfo.actionId !== actionId) {
+      this.actionInfo = await this.getActionInfo(actionId, namespace.namespace);
+    }
+
+    const task = taskInfo
+      ? {
+          task: {
+            scheduled: taskInfo.scheduled.toISOString(),
+            scheduleDelay: Millis2Nanos * (Date.now() - taskInfo.scheduled.getTime()),
+          },
+        }
+      : {};
+    // Write event log entry
+    const event = createActionEventLogRecordObject({
+      actionId,
+      consumer,
+      action: EVENT_LOG_ACTIONS.executeTimeout,
+      message: `action: ${this.actionInfo.actionTypeId}:${actionId}: '${
+        this.actionInfo.name ?? ''
+      }' execution cancelled due to timeout - exceeded default timeout of "5m"`,
+      ...namespace,
+      ...task,
+      executionId,
+      spaceId,
+      savedObjects: [
+        {
+          type: 'action',
+          id: actionId,
+          typeId: this.actionInfo.actionTypeId,
+          relation: SAVED_OBJECT_REL_PRIMARY,
+        },
+      ],
+      relatedSavedObjects,
+      actionExecutionId,
+      isInMemory: this.actionInfo.isInMemory,
+      ...(source ? { source } : {}),
+    });
+
+    eventLogger.logEvent(event);
+  }
+
+  public async getActionInfo(actionId: string, namespace: string | undefined): Promise<ActionInfo> {
```

Comment:
renamed from `getActionInfoInternal` because it was used externally in the task runner factory 🤷‍♀️